src/main/java/org/helpberkeley/memberdata/UploadFile.java

```diff
@@ -45,6 +49,22 @@ public final String getOriginalFileName() {
         return originalFileName;
     }
 
+    public static UploadFile createUploadFile(String data){
+        String fileName = HBParser.downloadFileName(data);
+        String shortURL = HBParser.shortURLDiscoursePost(data);
+        return new UploadFile(fileName, shortURL);
+    }
+
+    public static boolean auditFilePrefix (final String line) throws MemberDataException {
+        if (!(line.contains(Constants.UPLOAD_URI_PREFIX) || line.contains(Constants.WEB_CSV_PREFIX))) {
+            int prefixStart = line.lastIndexOf("]")+2;
+            int prefixEnd = line.lastIndexOf("/")+1;
```

Comment:
Remove this indexing.  It can crash if the user provided link is completely garbage and doesn't contain what you are expecting here.  Just used the entire line in the exception message.